packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -118,6 +118,10 @@ export class Header extends GOVUKFrontendComponent {
    * @private
    */
   syncState() {
+    if (!this.mql || !this.$menu || !this.$menuButton) {
+      return
+    }
```

Comment:
**question** I _think_ this will never happen:
* We early return if `this.$menu` is falsy
* We throw if `this.$menu` exists and `this.$menuButton` is falsy
* And if we haven't thrown, we create `this.mql` ourselves

What's requiring us to add this early return?

Reply:
Same as we [discussed via Slack thread again](https://gds.slack.com/archives/C05P6FJ1YH2/p1696235949838579?thread_ts=1695976340.212499&cid=C05P6FJ1YH2)

Don't forget that `this.$menu` being falsy causes `this.mql` to stay as `null`

Unlike in **Tabs**, the compiler spotted that `this.mql` isn't always set in the constructor

Can see all these with strict mode turned on in **packages/govuk-frontend/tsconfig.build.json**

```patch
{
  "extends": "../../tsconfig.base.json",
  "include": ["./src/govuk/**/*.mjs"],
  "exclude": ["**/*.test.*"],
  "compilerOptions": {
    "lib": ["ESNext", "DOM"],
+  "strict": true,
    "target": "ES2015",
    "types": ["node"]
  }
}
```

Reply:
As an alternative, I've mentioned [getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) in https://github.com/alphagov/govuk-frontend/pull/4323#discussion_r1357003285

Changing `this.mql` to a [getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) (without a [setter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set)) ensures it only runs when first accessed:

```mjs
get mql() {
  if (!this._mql) {
    this._mql = window.matchMedia('(min-width: 48.0625em)')
  }

  return this._mql
}
```